protocol/proofs/dispute-game-creators.md

```diff
@@ -0,0 +1,100 @@
+# Purpose
+
+The current fault-proof system requires a redeployment of the `DisputeGame.sol`, `DisputeGameFactory.sol`, and `AnchorStateRegistry.sol`. Due to chain-specific immutable variables (like `_weth`, `_anchorStateRegistry`, `_l2ChainId`) in its constructor. This necessitates redeploying associated `DisputeGameFactory.sol` and `AnchorStateRegistry.sol` contracts as well. This document proposes changes to simplify the Fault Dispute Game contract setup, enabling reuse of a single implementation across chains by externalizing chain-specific configuration.
```

Comment:
Should specify when it requires a redeployment (ie when deploying a new chain, and when upgrading). It's also worth calling out the different type of immutables (CWIA vs Immutables) the dispute game contracts have right now. It's also not so much as externalizing as it is relocating, they should still be contained somewhere vaguely attached to the dispute game